aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -623,6 +617,24 @@ void TensorIterator::mark_outputs() {
   }
 }
 
+void TensorIterator::check_overlaps() {
+  if (check_overlap_) {
+    for (int i = 0; i < num_outputs_; i++) {
+      const auto &output = operands_[i].tensor;
+      if (!output.defined()) continue;
```

Comment:
@VitalyFedyunin out of curiosity, is there ever a case where some outputs are defined but others aren't?

Reply:
I havn't seen any, but this check is a common pattern here, I think we should preserve it for now